src/tiled/propertiesdock.cpp

```diff
@@ -223,9 +223,15 @@ void PropertiesDock::removeProperty()
 
     const QString name = item->property()->propertyName();
     QUndoStack *undoStack = mMapDocument->undoStack();
+    if (item->parent()->children().count() > 1) {
+        int currentItemIndex = item->parent()->children().indexOf(item);
+        if (item == item->parent()->children().last()) {
+            mPropertyBrowser->setCurrentItem(item->parent()->children().at(currentItemIndex - 1));
+        } else {
+           mPropertyBrowser->setCurrentItem(item->parent()->children().at(currentItemIndex + 1));
```

Comment:
After seeing `item->parent()->children()` for the 5th time, it's really asking for the introduction of a local variable for it.